docs/howto/operate/add-aws-hub.md

```diff
@@ -0,0 +1,211 @@
+[Set up and use the the deployment scripts locally](./manual-deploy.md) to set up the
+local environment and prepare `sops` to encrypt and decrypt files.
+
+2. Install the awscli tool (you can use pip or conda to install it in the environment)
+and configure it to use the provided AWS user credentials. Follow this
+[guide](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html#cli-configure-quickstart-config)
+for a quick configuration process.
+
+```{note}
+The customer with AWS admin privileges should have created a user for you with full
+privileges. We will probably explore fine-graining the permissions actually needed
+in the short-term.
+```
+
+3. Export some helpful AWS environment variables (because "aws configure" doesn't export
+these environments vars for kops to use, so we do it manually).
+
+```bash
+export AWS_PROFILE=<cluster_name>
+export AWS_ACCESS_KEY_ID=$(aws configure get aws_access_key_id)
+export AWS_SECRET_ACCESS_KEY=$(aws configure get aws_secret_access_key)
+```
+
+## Create an AWS kops-based cluster
+
+1. From the root directory on this repo, `cd` into the `kops` directory
+2. Set up a *state* bucket for kops. This bucket will store the cluster
+[state](https://kops.sigs.k8s.io/state/).
+
+``` bash
+export KOPS_STATE_STORE=s3://<2i2c>-<hub-name>-kops-state
+aws s3 mb $KOPS_STATE_STORE --region <region>
+```
+
+```{note}
+Kops [recommends](https://kops.sigs.k8s.io/getting_started/aws/#cluster-state-storage)
+versioning the S3 bucket in case you ever need to revert or recover a previous state
+store.
+```
+
+### Create and render a kops config file
+
+You can use 
+[one](https://github.com/2i2c-org/pilot-hubs/blob/master/kops/farallon.jsonnet) of the
+existing [jsonnet](https://jsonnet.org/) specifications as a "template" for your cluster.
+You may need to tweak zones, names and instances, the rest is boilerplate to create a
+kops-based cluster accordingly to some specification already outlined in 
+[#28](https://github.com/2i2c-org/pangeo-hubs/issues/28). Once you have your jsonnet
+specification ready, you need to render it to create the config file kops understand.
+
+1. Render the `kops` config file with
+
+```bash
+jsonnet <cluster_name>.jsonnet -y > <cluster_name>.kops.yaml
+```
+
+2. Regrettably, the rendering creates yaml file with with 3 dots at the end, you can 
+delete it with
+
+```bash
+sed -i '' -e '$ d' <cluster_name>.kops.yaml
+```
+
+```{note}
+In Linux you will need to use instead: `sed -i '$ d' <cluster_name>.kops.yaml`
+```
+
+### Create the cluster
+
+1. Create the cluster configuration and push it to s3 with
+
+```bash
+kops create -f <cluster_name>.kops.yaml
+```
+
+2. You will need to a ssh key pair before actually creating the cluster
+
+```bash
+ssh-keygen -f ssh-key
+mv ssh-key <cluster_name>.key
+mv ssh-key.pub <cluster_name>.key.pub
+```
+
+```{note}
+You will need to relocate and encrypt the private key before pushing your changes to the
```

Comment:
Add commands for this too?

Reply:
Or we could add a separate section on encrypting private keys, since that is probably a generic thing and not AWS-specific, and cross-link it?

Reply:
Both options are OK... I think I would probably add the step here for now (I was thinking in that direction before pushing this one) and then we can add a new section and cross-reference in another PR...

Reply:
Added a new 4h step to relocate and encrypt the private key.